app/models/miq_search.rb

```diff
@@ -8,6 +8,16 @@ class MiqSearch < ApplicationRecord
 
   before_destroy :check_schedules_empty_on_destroy
 
+  # validate if the name of a new filter is unique in Global Filters
```

Comment:
From the beginning, I was trying to use similar idea, but I was not successful. I will try yours. Thanks!